Makefile

```diff
@@ -69,6 +69,7 @@ template:
 	sed -e 's/$$PGUSER/$(PGUSER)/' -e 's/$$PGPASS/$(PGPASS)/'  conf/db.conf.in  > conf/db.conf
 	cat conf/db.conf conf/*.part > conf/sphinx.conf
 	indextool --checkconfig -c conf/sphinx.conf | grep "config valid"
+	@[ -f .git/hooks/pre-commit ] || cp scripts/pre-commit.sh .git/hooks/pre-commit && chmod +x .git/hooks/pre-commit
```

Comment:
That's really fine, but ideal would be to watch, `scripts/pre-commit.sh` and when it changes trigger an update of the pre-commit hook.

if you do:
```
scripts/pre-commit.sh:   (e.g. watch for changes of this file)
.git/hooks/pre-commit: scripts/pre-commit.sh (trigger .git/hooks/pre-commit if dep on the right has changed)
  cp scripts/pre-commit.sh .git/hooks/pre-commit && chmod +x .git/hooks/pre-commit
```

Which can be translated to:
```
scripts/pre-commit.sh:
.git/hooks/pre-commit: scripts/pre-commit.sh
  cp $^ $@ && chmod +x $@
```
 `$@` and `$^` being the left and right sides respectively.

Then you add 
.git/hooks/pre-commit
as a dep to templates so that you can remove the checkconfig just above as well.

Untested code :smile: